photon-ml/src/main/scala/com/linkedin/photon/ml/data/KeyValueScore.scala

```diff
@@ -63,7 +63,12 @@ protected[ml] class KeyValueScore(val scores: RDD[(Long, Double)]) extends RDDLi
     val addedScores =
       this.scores
         .fullOuterJoin(that.scores)
-        .mapValues { case (thisScore, thatScore) => thisScore.getOrElse(0.0) + thatScore.getOrElse(0.0) }
+        .mapValues { case (thisScore, thatScore) =>
+          require(!thisScore.isEmpty || !thatScore.isEmpty, s"Should not have reached this state. A full outer join " +
```

Comment:
That was my first impression - but then maybe there is a special case where that could happen? If it does, it deserves a comment.